compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -1717,6 +1717,49 @@ dataDependencyTests Tools{damlc,repl,validate,davlDar,oldProjDar} = testGroup "D
             ]
         withCurrentDirectory (tmpDir </> "proj") $
             callProcessSilent damlc ["build"]
+
+    , testCaseSteps "Implicit parameters" $ \step -> withTempDir $ \tmpDir -> do
+        step "building project with implicit parameters"
+        createDirectoryIfMissing True (tmpDir </> "dep")
+        writeFileUTF8 (tmpDir </> "dep" </> "daml.yaml") $ unlines
+            [ "sdk-version: " <> sdkVersion
+            , "name: dep"
+            , "source: ."
+            , "version: 0.1.0"
+            , "dependencies: [daml-prim, daml-stdlib]"
+            ]
+        writeFileUTF8 (tmpDir </> "dep" </> "Foo.daml") $ unlines
+            [ "module Foo where"
+            , "f : Scenario ()"
+            , "f = scenario do"
+            , "  p <- getParty \"p\""
+            , "  submit p $ pure ()"
+            , "  submit p $ pure ()"
+            -- This will produce two implicit instances.
+            -- GHC occasionally seems to inline those instances and I don’t understand
+            -- how to reliably stop it from doing this so be careful when changing this.
+            ]
+        withCurrentDirectory (tmpDir </> "dep") $
+            callProcessSilent damlc ["build", "-o", "dep.dar"]
```

Comment:
I wonder if we can make a test that reliably generates an implicit parameter instance?

Or perhaps, add some a check to this test that an implicit parameter instance is in the DAR.

Reply:
Ah, I wonder if we can slip it in to the "Simple DALF" test or something. (It's not so simple...)

Reply:
I’ve added an assertion that we have 2 implicit instances. Simple DALF would work as well I guess but this has the advantage that it tests what our users actually hit which seems useful. I still don’t understand how to generate it reliably. Even in this example you can remove the type signature, the inferred type stays the same but now it’s inlined :shrug:

Reply:
Awesome, thank you!